src/image-viewer.js

```diff
@@ -490,23 +501,49 @@ export class ImageViewer {
   }
 
   /**
-   * Performs a one-step zoom action.
+ * Performs a one-step pinch zoom action.
+ * @param {number} centerClientX
+ * @param {number} centerClientY
+ * @param {number} deltaX
+ * @param {number} deltaY
+ *  @param {number} dir
+ * @private
+ */
+  onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir) {
+    this.onZoomInc_(centerClientX, centerClientY, deltaX, deltaY, dir);
+  }
+
+/**
+   * Performs a one-step tap zoom action.
    * @param {number} centerClientX
    * @param {number} centerClientY
    * @param {number} deltaX
    * @param {number} deltaY
    * @private
    */
-  onZoomInc_(centerClientX, centerClientY, deltaX, deltaY) {
-    const dist = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
 
-    const zoomSign = Math.abs(deltaY) > Math.abs(deltaX) ?
-        Math.sign(deltaY) : Math.sign(-deltaX);
-    if (zoomSign == 0) {
+  onTapZoom_(centerClientX, centerClientY, deltaX, deltaY) {
+    const dir = Math.abs(deltaY) > Math.abs(deltaX) ?
+      Math.sign(deltaY) : Math.sign(-deltaX);
+    this.onZoomInc_(centerClientX, centerClientY, deltaX, deltaY, dir);
```

Comment:
made me think of `Monster Inc` :). Mayeb a different name? `performZoom`, `applyZoom`?